src/lib/stats.rs

```diff
@@ -15,14 +22,80 @@ pub fn pearson_correlation_coefficient(xs: &Vec<f32>, ys: &Vec<f32>) -> f64{
 
     let lval: f64 = 1.0 / ((n as f64) - 1.0);
 
-    return rval * lval;
+    let final_val = rval * lval;
+
+    if final_val.is_nan() {
+        return 0.0;
+    }
+
+    return final_val;
 }
 
 pub fn pairinate(a: &BinaryTimeSeries, b: &BinaryTimeSeries)
-                 -> (Vec<f32>, Vec<f32>){
-    if a.data.len() != b.data.len(){
-        panic!("AHHHH! Everything is broken. Those datas don't match.");
+                 -> Option<(Vec<f32>, Vec<f32>)>{
+
+    let a_start = try_opt!(get_start_index(a, b));
+    let b_start = try_opt!(get_start_index(b, a));
+    let a_end = try_opt!(get_end_index(a, b));
+    let b_end = try_opt!(get_end_index(b, a));
+
+    if (a_end - a_start) >= (b_end - b_start) {
+        // Only one point of overlap, cannot correlate
+        if a_end - a_start == 0 {return None};
```

Comment:
Might want to change this to need at least three or four points of overlap. Two points increasing correlates with anything increasing that overlaps for only those two points. Not a big deal though since there won't be many tiny overlaps probably.